### PR TITLE
Add skeleton for wasmer-python

### PR DESCRIPTION
**Pull request by ibdafna**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by bollwyvl** [comment]
thanks @ibdafna! :star_struck:  

on behalf of @conda-forge/wasmer, will be happy to help on this.

File: recipes/wasmer-python/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{% set version = "1.1.0" %}
+
+package:
+  version: {{ version }}
```
Comment (bollwyvl) [inline recipes/wasmer-python/meta.yaml:4]: needs 
```
package:
  name: wasmer-python
```

File: recipes/wasmer-python/build.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+cargo install just
```
Comment (bollwyvl) [inline recipes/wasmer-python/build.sh:1]: needed if we have it?

Also it's good to preamble with somehting like:

```bash
#!/usr/bin/env bash
set -eux
```

File: recipes/wasmer-python/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    - maturin
+    - just
+    - cargo
+  run:
+    - python
+
+test:
+  imports:
+    - wasmer
+    - wasmer-compiler-cranelift
```
Comment (bollwyvl) [inline recipes/wasmer-python/meta.yaml:32]: these get translated into e.g.

```
python -c "import wasmer-compiler-cranelift"
```

soe the thing with `-` isn't going to import.

**Comment by ibdafna** [comment]
@bollwyvl sorry for the PR spam!!

**Comment by bollwyvl** [comment]
Also, taking a look at the docs: we're going to probably want multiple outputs:

- `wasmer-python`
- `wasmer-python-compiler-singlepass`
- `wasmer-python-compiler-cranelift`
- `wasmer-python-compiler-llvm`

The motivation here is that some downstreams will be using this to just ship some pre-compiled WASM used inside of a library, and shouldn't require pulling in a full compiler.

**Comment by wolfv** [comment]
Hi @ibdafna happy to help as well! :)

I am wondering if we need to run the `prelude`: https://github.com/wasmerio/wasmer-python/blob/31fae063239b8cc4d785b153c58693fc1feba2f1/justfile#L2-L19 
This currently fails because we don't have virtualenv available. However, we don't really need virtualenv since we already have a conda env. On the other hand I don't know maturin much, and it might indeed require virtualenv.

**Comment by bollwyvl** [comment]
Seems promising... We have other maturin builds that don't don't use
virtualenv...

**Comment by ibdafna** [comment]
Hey @wolfv 👋 excited to see you here - thank you for the help!

You're right with regards to not running the prelude - I was able to get something to build internally without running the prelude beforehand. I'm going to update the PR!

**Comment by wolfv** [comment]
I've added a modified `justfile`. lets see if that helps! I removed their detection of `PYTHON_SYS_EXECUTABLE`.

**Pull request closed without merging**